### PR TITLE
[FLAVA]Use the ckpt loading helper function instead of PretrainedMixin

### PR DESCRIPTION
**Pull request by ankitade**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #368
* __->__ #366
* #365

Every class has to start inheriting PretrainedMixin. Instead its easier to just call the helper function 

Test plan
1. Pretrain debug
python -m flava.train config=flava/configs/pretraining/debug.yaml training.lightning.gpus=0 training.lightning.strategy=null
Epoch 0: : 50it [05:00,  6.01s/it, loss=12.1, v_num=5, train/losses/mmm_text_loss=9.870, train/losses/mmm_image_loss=9.100, train/losses/itm_loss=0.383, train/losses/global_contrastive_loss=2.100, train/losses/mlm_loss=10.20, train/losses/mim_loss=9.190]

2. Finetune 
python -m flava.finetune config=flava/configs/finetuning/qnli.yaml training.lightning.accelerator=cpu training.lightning.gpus=0 training.lightning.strategy=null
Epoch 0:   1%|▊                                                                 | 50/3787 [31:28<39:11:53, 37.76s/it, loss=0.713, v_num=6, train/losses/classification=0.697, train/accuracy/classification=0.469]

3. zero shot
python -m flava.coco_zero_shot --data_root /datasets01/COCO/022719/val2017 --annotations /datasets01/COCO/022719/annotations/captions_val2017.json

Differential Revision: [D41142482](https://our.internmc.facebook.com/intern/diff/D41142482)

### CODE REVIEW

**Comment by ankitade** [comment]
@deankita has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D41142482).

**Comment by ankitade** [comment]
@deankita has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D41142482).

**Comment by ankitade** [comment]
@deankita has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D41142482).

File: torchmultimodal/models/flava/model.py [hunk 1]
```diff
@@ -714,10 +716,11 @@ def __init__(
...
     def load_model(self) -> Any:  # type: ignore
         # TODO (T116682215): Network error due to FLAVA model relying on access to openAI
-        encoder = super().load_model(
-            "https://cdn.openai.com/dall-e/encoder.pkl", load_state_dict=False
+
+        encoder_state_dict = torch.hub.load_state_dict_from_url(
+            "https://cdn.openai.com/dall-e/encoder.pkl"
         )
-        self.encoder.load_state_dict(encoder.state_dict())  # type: ignore
+        self.encoder.load_state_dict(encoder_state_dict.state_dict())  # type: ignore
```
Comment (ebsmothers) [inline torchmultimodal/models/flava/model.py:723]: wait why are we calling .state_dict() on something that's already a state_dict?

Comment (ankitade) [reply torchmultimodal/models/flava/model.py:723]: its the same as before

Comment (ankitade) [reply torchmultimodal/models/flava/model.py:723]: https://colab.research.google.com/drive/1adoRcJtQWq-zVQuVwyO-4m5Qfx6hlILx#scrollTo=0X0QoOwnV1Ei
